### PR TITLE
Add support for named columns in CSV.

### PR DESCRIPTION
**Pull request by berkus**
Make CSV line arrays use column names as keys. Simplifies inserting into
databases or just accessing CSV columns by name.

### CODE REVIEW

File: src/Goodby/CSV/Import/Standard/LexerConfig.php [hunk 1]
```diff
@@ -182,4 +187,23 @@ public function getIgnoreHeaderLine()
...
+    public function setUseNamedColumns($useNamedColumns)
+    {
+        $this->useNamedColumns = $useNamedColumns;
+        return $this;
+    }
+
+    /**
+     * @return boolean
+     */
+    public function getUseNamedColumns()
```
Comment (suin) [inline src/Goodby/CSV/Import/Standard/LexerConfig.php]: __[phpmd]__
* The 'getUseNamedColumns()' method which returns a boolean should be named 'is...()' or 'has...()'

File: src/Goodby/CSV/Import/Standard/Lexer.php [hunk 2]
```diff
@@ -60,6 +61,10 @@ public function parse($filename, InterpreterInterface $interpreter)
             if ($ignoreHeader && $lineNumber == 0 || (count($line) === 1 && trim($line[0]) === '')) {
                 continue;
             }
+            else if ($namedColumns && $lineNumber == 0) {
```
Comment (suin) [inline src/Goodby/CSV/Import/Standard/Lexer.php]: __[code_sniffer]__
* Usage of ELSE IF is discouraged; use ELSEIF instead

File: src/Goodby/CSV/Import/Standard/LexerConfig.php (hunk 1 above)
Comment (berkus) [inline src/Goodby/CSV/Import/Standard/LexerConfig.php]: Not applicable, as this is mimicking already existing code structure. See `getIgnoreHeaderLine()` above for example.

**Comment by zackkatz** [comment]
:+1:

**Comment by garygreen** [comment]
@suin any news on this? Would be much welcome addition 😄

**Pull request closed without merging**